apps/allocations/app/components/Card/Budget.js

```diff
@@ -0,0 +1,213 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import styled from 'styled-components'
+import { BigNumber } from 'bignumber.js'
+
+import {
+  Card,
+  ContextMenu,
+  ContextMenuItem,
+  IconEdit,
+  IconPlus,
+  IconProhibited,
+  IconView,
+  ProgressBar,
+  Text,
+  useTheme,
+} from '@aragon/ui'
+import { displayCurrency } from '../../utils/helpers'
+
+import { BASE_CARD_WIDTH, CARD_STRETCH_BREAKPOINT } from '../../utils/responsive'
+
+const Budget = ({
+  id,
+  name,
+  amount,
+  currency,
+  allocated,
+  inactive,
+  onNewAllocation,
+  onEdit,
+  onDeactivate,
+  onReactivate,
+  screenSize,
+}) => {
+  const theme = useTheme()
+  const newAllocation = () => {
+    onNewAllocation(id, name, amount)
+  }
+  const edit = () => {
+    onEdit(id)
+  }
+  const deactivate = () => {
+    onDeactivate(id)
+  }
+
+  const reactivate = () => {
+    onReactivate(id)
+  }
+
+  if (inactive) {
+    return (
+      <StyledCard screenSize={screenSize}>
+        <MenuContainer>
+          <ContextMenu>
+            <ContextMenuItem onClick={reactivate}>
+              <IconView />
+              <ActionLabel>Reactivate</ActionLabel>
+            </ContextMenuItem>
+          </ContextMenu>
+        </MenuContainer>
+        <CardTitle theme={theme}>{name}</CardTitle>
+        <StatsContainer>
+          <StyledStats>
+            <StatsValueBig theme={theme}>
+              <Text>Inactive</Text>
+            </StatsValueBig>
+          </StyledStats>
+        </StatsContainer>
+      </StyledCard>
+    )
+  }
+
+  return (
+    <StyledCard screenSize={screenSize}>
+      <MenuContainer>
+        <ContextMenu>
+          <ContextMenuItem onClick={newAllocation}>
+            <IconPlus />
+            <ActionLabel>New Allocation</ActionLabel>
+          </ContextMenuItem>
+          <ContextMenuItem onClick={edit}>
+            <IconEdit />
+            <ActionLabel>Edit</ActionLabel>
+          </ContextMenuItem>
+          <ContextMenuItem onClick={deactivate}>
+            <IconProhibited />
+            <ActionLabel>Deactivate</ActionLabel>
+          </ContextMenuItem>
+        </ContextMenu>
+      </MenuContainer>
+      <CardTitle theme={theme}>{name}</CardTitle>
+      <StatsContainer>
+        <StyledStats>
+          <StatsValueBig theme={theme}>
+            {displayCurrency(BigNumber(amount))}
+            <Text>{' ' + currency + ' per period'}</Text>
+          </StatsValueBig>
+          <StatsValueBig css={{ paddingTop: '24px' }} theme={theme}>
+            <ProgressBar
+              color={String(theme.accentEnd)}
+              value={BigNumber(allocated).div(amount).toNumber()}
+            />
+          </StatsValueBig>
+          <StatsValueSmall css={{
+            color: theme.content,
+            paddingTop: '8px',
+          }}>
+            {displayCurrency(BigNumber(amount).minus(allocated))}
```

Comment:
This should be just an integer, or even a string, but not the result of a function called on the fly during render lifecycle